src/index.ts

```diff
@@ -1,178 +1,269 @@
-// Generated data
-import {
-  GPU_BENCHMARK_SCORE_DESKTOP,
-  GPU_BENCHMARK_SCORE_MOBILE,
-} from './__generated__/GPUBenchmark';
+// Vendor
+import leven from 'leven';
+import fetch from 'unfetch';
 
 // Internal
-import { cleanEntryString } from './internal/cleanEntryString';
-import { cleanRendererString } from './internal/cleanRendererString';
-import { getBenchmarkByPercentage } from './internal/getBenchmarkByPercentage';
-import { browser, isMobile, isTablet } from './internal/getBrowserType';
-import { getEntryVersionNumber } from './internal/getEntryVersionNumber';
-import { getWebGLUnmaskedRenderer } from './internal/getWebGLUnmaskedRenderer';
-import { isWebGLSupported } from './internal/isWebGLSupported';
-import { getLevenshteinDistance } from './internal/getLevenshteinDistance';
+import { getGPUVersion } from './internal/getGPUVersion';
+import { getWebGLContext } from './internal/getWebGLContext';
+import { deviceInfo, isSSR } from './internal/deviceInfo';
+import { deobfuscateRenderer } from './internal/deobfuscateRenderer';
 
 // Types
-export interface IGetGPUTier {
-  glContext?: WebGLRenderingContext | WebGL2RenderingContext;
-  mobileBenchmarkPercentages?: number[];
-  desktopBenchmarkPercentages?: number[];
-  failIfMajorPerformanceCaveat?: boolean;
-  forceRendererString?: string;
-  forceMobile?: boolean;
-}
-
-export type TRank = [number, string] | [undefined, undefined];
-
-export interface IRankWithDistance {
-  rank: TRank;
-  distance: number;
-}
-
-export const getGPUTier = ({
-  mobileBenchmarkPercentages = [
-    0, // TIER_0
-    50, // TIER_1
-    30, // TIER_2
-    20, // TIER_3
-  ],
-  desktopBenchmarkPercentages = [
-    0, // TIER_0
-    50, // TIER_1
-    30, // TIER_2
-    20, // TIER_3
-  ],
-  forceRendererString = '',
-  forceMobile = false,
+import type {
+  GetGPUTier,
+  ModelEntry,
+  TierResult,
+  TierType,
+  ModelEntryScreen,
+} from './types';
+
+const debug = false ? console.log : undefined;
+
+const queryCache: { [k: string]: Promise<ModelEntry[] | undefined> } = {};
+
+export const getGPUTier = async ({
+  mobileTiers = [0, 15, 30, 60],
+  desktopTiers = [0, 15, 30, 60],
+  override: {
+    renderer,
+    isIpad = Boolean(deviceInfo?.isIpad),
+    isMobile = Boolean(deviceInfo?.isMobile),
+    screenSize = isSSR ? { height: 1080, width: 1920 } : window.screen,
+    loadBenchmarks,
+  } = {},
   glContext,
   failIfMajorPerformanceCaveat = true,
-}: IGetGPUTier = {}): { tier: string; type: string } => {
-  let renderer: string;
-  const isMobileTier = isMobile || isTablet || forceMobile;
-
-  const createGPUTier = (
-    index: number = 1,
-    GPUType: string = 'FALLBACK'
-  ): { tier: string; type: string } => ({
-    tier: `GPU_${isMobileTier ? 'MOBILE' : 'DESKTOP'}_TIER_${index}`,
-    type: GPUType,
-  });
+  benchmarksURL = '/benchmarks',
+}: GetGPUTier = {}): Promise<TierResult> => {
+  const MODEL_INDEX = 0;
```

Comment:
@puckey is there a reason why MODEL_INDEX is split out to a const like that?

Reply:
Just to avoid some magic numbers when accessing arrays by index..